cabal-install/src/Distribution/Client/CmdIde.hs

```diff
@@ -0,0 +1,126 @@
+import Distribution.Verbosity
+         ( normal )
+
+import Distribution.Client.TargetSelector
+
+
+-------------------------------------------------------------------------------
+-- Command
+-------------------------------------------------------------------------------
+
+ideCommand :: CommandUI (NixStyleFlags IdeFlags)
+ideCommand = CommandUI
+  { commandName = "ide"
+  , commandSynopsis = "Query for simple project information"
+  , commandDescription  = Just $ \_ -> wrapText $
+      "Query for available targets and project information such as project GHC."
+  , commandNotes = Nothing
+  , commandUsage = \pname ->
+      "Usage: " ++ pname ++ " ide [FLAGS]\n"
+  , commandDefaultFlags = defaultNixStyleFlags defaultIdeFlags
+  , commandOptions      = nixStyleOptions $ \showOrParseArgs ->
+      [ option [] ["targets"]
+        "List all available targets in the project"
+        ideTargets (\v flags -> flags { ideTargets = v })
+        (yesNoOpt showOrParseArgs)
+      , option [] ["project-compiler"]
+        "Print information of the project compiler"
+        ideProjectGhc (\v flags -> flags { ideProjectGhc = v })
+        (yesNoOpt showOrParseArgs)
+      ]
+  }
+
+-------------------------------------------------------------------------------
+-- Flags
+-------------------------------------------------------------------------------
+
+data IdeFlags = IdeFlags
+  { ideTargets    :: Flag Bool
+  , ideProjectGhc :: Flag Bool
+  }
+
+defaultIdeFlags :: IdeFlags
+defaultIdeFlags = IdeFlags
+  { ideTargets    = mempty
+  , ideProjectGhc = mempty
+  }
+
+
+-------------------------------------------------------------------------------
+-- Action
+-------------------------------------------------------------------------------
+
+-- | Entry point for the 'ide' command.
+ideAction :: NixStyleFlags IdeFlags -> [String] -> GlobalFlags -> IO ()
+ideAction flags@NixStyleFlags { extraFlags = ideFlags, ..} _targetStrings globalFlags = do
+
+  ProjectBaseContext {
+    distDirLayout,
+    cabalDirLayout,
+    projectConfig,
+    localPackages
+  } <- establishProjectBaseContext verbosity cliConfig OtherCommand
+
+  when (fromFlagOrDefault False (ideProjectGhc ideFlags)) $ do
+    (_, _, elabSharedConfig, _, _) <-
+      rebuildInstallPlan verbosity
+                          distDirLayout cabalDirLayout
+                          projectConfig
+                          localPackages
+
+    let progDb = pkgConfigCompilerProgs elabSharedConfig
+    (compilerProg, _) <- requireProgram verbosity ghcProgram progDb
+    putStrLn $ unlines
+      [ "Compiler: " ++ programId compilerProg
+      , "Version: " ++ maybe "UNKNOWN" prettyShow (programVersion compilerProg)
+      , "Path: " ++ programPath compilerProg
+      ]
+
+  when (fromFlagOrDefault False (ideTargets ideFlags)) $ do
+    putStrLn "Available Targets:"
+    knownTargets <- getKnownTargets defaultDirActions localPackages
+    for_ (knownComponentsAll knownTargets) $ \KnownComponent{..} -> do
```

Comment:
This impl is wrong, but getting it right is actually non-trivial.

Reply:
Any chance you can document the expected result so that it can be undertaken in another PR? :)

Reply:
This will be put on "maybe later" pile, so ignoring for now.